src/server.ml

```diff
@@ -76,7 +76,10 @@ module Make (IO : IO) = struct
     end
 
   let async (self : #base_server) f : unit IO.t =
-    self#spawn_query_handler f;
+    self#spawn_query_handler
+      (fun () -> try f () with err ->
```

Comment:
format? ~~but looks good~~ :)

wait we need to use the async catch, `f()` is async.